lib/openshift/report.rb

```diff
@@ -1,46 +1,69 @@
 # for reference on how to use Metering please consult the following link
 # https://github.com/operator-framework/operator-metering/blob/master/Documentation/using-metering.md
+
+require 'openshift/flakes/conditions'
+
 module BushSlicer
   class Report < ProjectResource
     RESOURCE = "reports"
 
-    def finished?(user: nil, cached: false, quiet: false)
-      phase(user:user, cached: cached, quiet: quiet) == :finished
+
+    def conditions(user: nil, cached: false, quiet: false)
+      rr = raw_resource(user: user, cached: cached, quiet: quiet).dig('status', 'conditions')
+      rr.map { |cond| Conditions.new cond }
+    end
+
+    def running?(user: nil, cached: false, quiet: false)
+      # the reason 'Scheduled' is an initial state what we
+      conditions.any? { |c| c.type == 'Running' and c.reason != 'Scheduled' }
     end
 
     # wait until the stauts of the Report becomes 'Finished'
-    def wait_till_finished(user: nil, quiet: false)
+    def wait_till_finished(user: nil, cached: false, quiet: false)
+      seconds = 60 # for PVs it can take longer than 30s
+      success = wait_for(seconds) do
+        finished?(user:user, cached: cached, quiet: quiet)
+      end
+      raise "Report '#{self.name}' didn't become :finished" unless success
+      return success
+    end
+
+    # for scheduledreport or a report that has not reached the reportingEnd
+    # time, we just need to check that the type is Running and
+    def wait_till_running(user: nil, cached: false, quiet: false)
       seconds = 60 # for PVs it can take longer than 30s
       success = wait_for(seconds) do
-        finished?(user:user, cached: false, quiet: quiet)
+        running?(user: user, cached: cached, quiet: quiet)
       end
-      raise "Report didn't become :finished" unless success
+      raise "Report didn't become :running" unless success
       return success
     end
 
-    # need 4 inputs
-    # a. ReportGenerationQuery
-    # b. reportingStart
-    # c. reportingEnd
-    # d. runImmediately
-    def self.generate_yaml(query_type: self.name, start_time: nil, end_time: nil, run_now: "true")
-      # span the whole year if user didn't give a range
-      start_time ||= "#{Time.now.year}" + "-01-01T00:00:00Z"
-      end_time ||= "#{Time.now.year}" + "-12-30T23:59:59Z"
-      run_immediately ||= "true"
+    def finished?(user: nil, cached: false, quiet: false)
+      ## TODO: make this backwardcompatible
```

Comment:
backward compatible to what?

Reply:
For older version of `operator-metering` the finished condition represented by a field `phase` which is gone for newer version.  It's just a nice to have, but since `metering` is still categorized as `beta`, backward compatibility is probably low priority.